lib/trailblazer/endpoint/options.rb

```diff
@@ -77,6 +77,7 @@ def self.add(normalizer, directive_name, options)
 
       def self.CallDirective(callable, option_name)
         ->((ctx, flow_options), *) {
+          ctx ||= {}
```

Comment:
Using `||=` usually implies you have lost track of how your data flows, it's always an anti-pattern for me! In what cases is that required for you, if I may ask?

Also, off-topic, @n-rodriguez may I ask you some questions on https://trailblazer.zulipchat.com of how you use `endpoint`? Background is, I have another branch where I threw out a lot of complex logic of this gem and changed the runtime API and I'd like to hear a user's thoughts (and so far, you're the only user I know hahaha). :coffee:

Reply:
> In what cases is that required for you, if I may ask?

For debugging :)

When I run this code without the patch :

```ruby
before_fork do
  PumaSettings.info '* Finalizing master process'

  #puts CampingsController.instance_variable_get('@normalizers')[:endpoints].call([]).inspect
  puts ConcertoFrance::PagesController.instance_variable_get('@normalizers')[:endpoints].call([]).inspect

  3.times { GC.start }
  GC.compact

  #puts CampingsController.instance_variable_get('@normalizers')[:endpoints].call([]).inspect
  puts ConcertoFrance::PagesController.instance_variable_get('@normalizers')[:endpoints].call([]).inspect
end
```

it fails with this error :

```sh

/Users/nicolas/PROJECTS/CONCERTO/concerto/.bundle/ruby/3.3.0/bundler/gems/trailblazer-endpoint-827ac62bdb4c/lib/trailblazer/endpoint/options.rb:80:in `block in CallDirective': no implicit conversion of nil into Hash (TypeError)

          config = callable.(ctx, **ctx) # e.g. ApplicationController.options_for_endpoint
```

Reply:
With this patch we can see the Ruby GC bug : https://github.com/trailblazer/trailblazer-activity/issues/60#issuecomment-2081158620